### DISCUSSION TITLE
Temporäres Laden mit max Ladung

### DISCUSSION THREAD

**Discussion by RienSte** [wiggal/GEN24_Ladesteuerung#85] [Ideas]
Guten Tag,

folgende Situation hatte ich letztens. Es war ein klassischer On/Off-Tag den die Prognose so natürlich nicht erfassen konnte. Aufgrund der Befürchtung der Akku könnte aufgrund der Ladesteuerung nicht voll werden habe ich die Ladung manuell auf VOLL gestellt (im Tab Ladesteuerung):
![image](https://github.com/user-attachments/assets/954443ea-6e19-4eee-9b8c-dd870ff34b19)

Mein Fehler war, dass ich am Abend vergessen hatte das wieder zurück zu stellen und somit die Ladung auf VOLL blieb. Ich war dann aber ein paar Tage nicht im Heimnetzwerk und von "außen" komme ich nicht - oder nur umständlich per Remote auf die Synology - auf die Steuerung. Somit wurde die Tage danach immer max. geladen und der Akku war dann bereits um 11:00 voll.

Macht zwar nix kaputt aber schön ist es auch nicht :-) da kam mir die Idee für eine Funktion die die Ladesteuerung nur temporär für 1, 2 , 3, n Tage pausiert und dann wieder auf AUTO zurückstellt.

Was hältst du davon?

SG Stephan

**Comment by wiggal** [comment]
Servus,

> ... Ich war dann aber ein paar Tage nicht im Heimnetzwerk und von "außen" komme ich nicht -.....

Für solche Fälle habe ich auf meiner Fritzbox ein VPN konfiguriert, ist super praktisch.

> Macht zwar nix kaputt aber schön ist es auch nicht :-) da kam mir die Idee für eine Funktion die die Ladesteuerung nur temporär für 1, 2 , 3, n Tage pausiert und dann wieder auf AUTO zurückstellt.

War eigentlich so konzipiert, dass man das z.B. im Winter nicht jeden Tag neu einstellen muss.
Eine Möglichkeit wäre evtl. ein Optionsschalter `nur für heute` oder so ähnlich, oder ein `Dropdown mit immer, 1, 2, 3 Tage`?

Oder hast du eine andere Idee?

Wiggal

**Reply by RienSte** [reply]
Das Thema VPN muss ich mir mal anschauen, wie gesagt ich komme grundsätzlich eh von außen rein. Halt direkt auf die DiskStation, auf der meine Ubuntu VM läuft mit deiner Steuerung. Aber das ist dann recht zäh, gibt ne massive Latenz mit der Maus und am Smartphone gehts gar nicht.

Ja so eine Art Zusatzoption wäre glaub ich schon ausreichend. Dropdown mit ein paar Möglichkeiten wär super 👍 Im Winter stelle ich mir das so vor, dass ich deine Steuerung so konfiguriere dass die die schlechte Prognose abfängt. Aber ich hatte noch keinen Winter mit Akku, also da hab ich keine Erfahrungen.

Aber ja, mit einer Option hätte man dann eh alle Möglichkeiten. Jetzt das ganze irgendwie wieder prognosebasiert (bspw. Wenn Prognose > 2 * (BattKapa + Grundlast) auf AUTO) machen birgt wieder Risiken. Und wenn man schon einen manuellen Mode einbaut, sollte der imho wirklich manuell sein und sich nicht wieder aufgrund irgendeiner (nicht direkt ersichtlichen) Automatik ausschalten (außer i.d.F. zeitbasiert natürlich, das ist ja für jeden klar nachvollziehbar).

**Reply by wiggal** [reply]
OK, ich nehme es mal bei Ideen auf

**Comment by iReckyy** [comment]
Bei mir funktioniert das aktuell garnicht. Wenn ich "Voll" stelle und schreibt er ganz normal weiter als ob es auf Auto stehen würde.
Unter Settings habe ich mit "unverändert lassen " und "WR Steuerung und Jogging" getestet.

Habe das System schon komplett neu gestartet, die Version ist aktuell (0.22.1)

Woran kann das liegen?

**Reply by wiggal** [reply]
> Bei mir funktioniert das aktuell garnicht. Wenn ich "Voll" stelle und schreibt er ganz normal weiter als ob es auf Auto stehen würde. Unter Settings habe ich mit "unverändert lassen " und "WR Steuerung und Jogging" getestet.

Was soll den funktionieren?

Ohne Logfileauszug kann ich da gar nichts sagen.

**Reply by iReckyy** [reply]
naja, er soll doch die Ladeleistung auf das in der config definierte Limit stellen, oder?
.
.
.
Ladesteuerung steht auf voll, das Limit ist 3000W, Schreibgrenze 500W.
.
.
.
.
***** BEGINN: 2024-07-26 15:26:01 *****

## HTTP-LADESTEUERUNG ##
>>>Parameteränderung durch WebUI-Settings: WR-Steuerung und Logging
aktuellePrognose: 2272
TagesPrognose-BattVollUm: 4737 - 17
PrognoseAbzugswert/Stunde: 439
Grundlast_Summe für Tag: 1096
aktuellePVProduktion/Watt: 419
aktuelleEinspeisung/Watt: 45
aktuelleBatteriePower/Watt: 40
GesamtverbrauchHaus/Watt: 414
aktuelleBattKapazität/Watt: 3815
Batteriestatus in Prozent: 47.7 %
LadewertGrund: Prognoseberechnung
Bisheriger Ladewert/Watt: 2238
Neuer Ladewert/Watt: 2749

CHARGE_MAX geschrieben: 2749W

In SQLite-Datei gespeichert!
***** ENDE: 2024-07-26 15:26:02 *****

**Reply by wiggal** [reply]
Ja er sollte auf 3000 stellen, und folgendes melden.
`LadewertGrund:  Manuelle Ladesteuerung in PV-Planung ausgewählt.`
Was steht den in der Steuerdatei `Watt_Reservierung.json` in der letzten Zeile?
Hier muss `"ManuelleSteuerung": 1` stehen, nicht `"ManuelleSteuerung": 0`

**Reply by iReckyy** [reply]
Da steht eine 1 oder 0 drin wenn ich zwischen Auto oder Voll wechsele.

Trotzdem gibt er mir bei Voll folgendes raus:
.
.
.
***** BEGINN: 2024-07-26 16:04:07 *****

## HTTP-LADESTEUERUNG ##
>>>Parameteränderung durch WebUI-Settings: WR-Steuerung und Logging
aktuellePrognose: 2782
TagesPrognose-BattVollUm: 3487 - 17
PrognoseAbzugswert/Stunde: 255
Grundlast_Summe für Tag: 653
aktuellePVProduktion/Watt: 5769
aktuelleEinspeisung/Watt: 623
aktuelleBatteriePower/Watt: -3719
GesamtverbrauchHaus/Watt: 1427
aktuelleBattKapazität/Watt: 3232
Batteriestatus in Prozent: 55.7 %
LadewertGrund: TagesPrognoseGesamt - Grundlast_Summe < BattKapaWatt_akt
Bisheriger Ladewert/Watt: 4000
Neuer Ladewert/Watt: 4000

Änderung kleiner Schreibgrenze!

In SQLite-Datei gespeichert!
***** ENDE: 2024-07-26 16:04:07 *****

**Reply by wiggal** [reply]
Also Limit steht auf 4000, oder?
Du hast bei Umstieg die neuen Konfigdateien `charge_priv.ini  default_priv.ini  weather_priv.ini` erzeugt, oder?
Hast du in der  `charge_priv.ini` `PV_Reservierung_steuern = 1` gestellt?
Sonst fällt mir nichts mehr ein.

**Reply by iReckyy** [reply]
Ja, habe das Limit auf 4000 gestellt. Die Priv Dateien habe ich erstellt.
Mist, das war es, ist mir durch die Lappen gegangen.

Danke.

**Comment by wiggal** [comment]
> folgende Situation hatte ich letztens. Es war ein klassischer On/Off-Tag den die Prognose so natürlich nicht erfassen konnte. Aufgrund der Befürchtung der Akku könnte aufgrund der Ladesteuerung nicht voll werden habe ich die Ladung manuell auf VOLL gestellt (im Tab Ladesteuerung)
> Mein Fehler war, dass ich am Abend vergessen hatte das wieder zurück zu stellen und somit die Ladung auf VOLL blieb. Ich war dann aber ein paar Tage nicht im Heimnetzwerk und von "außen" komme ich nicht - oder nur umständlich per Remote auf die Synology - auf die Steuerung. Somit wurde die Tage danach immer max. geladen und der Akku war dann bereits um 11:00 voll.
> 
> Macht zwar nix kaputt aber schön ist es auch nicht :-) da kam mir die Idee für eine Funktion die die Ladesteuerung nur temporär für 1, 2 , 3, n Tage pausiert und dann wieder auf AUTO zurückstellt.
> 
> Was hältst du davon?

OK, kommen wird nochmal darauf zurück, inzwischen hat sich ja einiges verändert am Tool, und es könnte evtl. jetzt einfacher eingebaut werden.
In der `LadeStrg` haben wir ja nun ein Dropdown mit den Einträgen `Auto, Slider, MaxLadung` man könnte nun bei Auswahl von Slider oder MaxLadung beim Speichern die Anzahl der Stunden abfragen, die die Auswahl gültig sein sollen, z.B.:

![grafik](https://github.com/user-attachments/assets/71bbc7ca-4336-40e6-95b3-940acc91f2fd)

Dann speichert man den Unixstamp und nach diesem ist wieder Auto gültig.
Ist nur mal eine Idee, müsste man noch vertiefen.

**Reply by RienSte** [reply]
Hi @wiggal 

ja, das schaut schon gut aus. Ich glaub, mehr brauchts auch gar nicht :-) Danke!
Ich schau mir das an und melde mich, falls noch was unzureichend ist.

SG Stephan

**Reply by iReckyy** [reply]
> OK, kommen wird nochmal darauf zurück, inzwischen hat sich ja einiges verändert am Tool, und es könnte evtl. jetzt einfacher eingebaut werden. In der `LadeStrg` haben wir ja nun ein Dropdown mit den Einträgen `Auto, Slider, MaxLadung` man könnte nun bei Auswahl von Slider oder MaxLadung beim Speichern die Anzahl der Stunden abfragen, die die Auswahl gültig sein sollen, z.B.:
> 
> ![grafik](https://private-user-images.githubusercontent.com/69506553/447545145-71bbc7ca-4336-40e6-95b3-940acc91f2fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg2ODAzMjQsIm5iZiI6MTc0ODY4MDAyNCwicGF0aCI6Ii82OTUwNjU1My80NDc1NDUxNDUtNzFiYmM3Y2EtNDMzNi00MGU2LTk1YjMtOTQwYWNjOTFmMmZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTMxVDA4MjcwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxODllNjExODU2ZWRiZDVmZTBlNzc5OWFhZDgwZDVkNmFmYzY1N2IyNzhhYWZiZGU2Zjg4YWI1MmQzZWNlYWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iCSAiD3C5dblPbRY-zi3moAALSNm43mV1YiHIfaPnuI)
> 
> Dann speichert man den Unixstamp und nach diesem ist wieder Auto gültig. Ist nur mal eine Idee, müsste man noch vertiefen.


Ich habe, glaube ich, ausersehen einen Bug gefunden.
>
Habe gestern geupdatet und mir die Änderungen angeschaut, dann habe ich mal auf Max Ladung gestellt, gespeichert, das Eingabefeld kam, habe aber nichts eingegeben. Ich weiß nicht mehr ob ich abbrechen gedrückt habe, oder die Seite einfach geschlossen, jedenfalls hat das die Steuerung komplett lahm gelegt.
>
[Uploading Crontab 4.log…]()

**Reply by wiggal** [reply]
Kann ich nicht nachvollziehen.
was meinst du mit "komplett lahm gelegt"??

**Reply by wiggal** [reply]
OK, ich glaube, ich habe es gefunden. Nach Eingabe von Buchstaben funktioniert das Prog nicht mehr.
Habe es gefixed: #185

**Reply by iReckyy** [reply]
> OK, ich glaube, ich habe es gefunden. Nach Eingabe von Buchstaben funktioniert das Prog nicht mehr. Habe es gefixed: #185

Eigentlich hatte ich garnichts eingegeben.
>
Hatte nur geschaut ob das Fenster kommt.
>
Mit komplett lahmgelegt meine ich, dass keine QZ-Statistik erstellt wurde und im Log nichts stand, ob es geschrieben hat oder nicht, kann ich nicht mehr sagen.

**Reply by wiggal** [reply]
Wenn du nichts eingibst, müsste die Vorbelegung `24` gespeichert werden.
Auch bei einem Abbruch usw. konnte ich keinen Fehler finden, einzig bei Eingabe von Buchstaben.

> Mit komplett lahmgelegt meine ich, dass keine QZ-Statistik erstellt wurde und im Log nichts stand, ob es geschrieben hat oder nicht, kann ich nicht mehr sagen.

Die QZ-Statistik dürfte davon gar nicht betroffen sein. Wenn du das nochmal reproduzieren kannst, schreib mir bitte.

**Reply by iReckyy** [reply]
Ja, mache ich
